bundles/http/Citrix/Microsoft-Outlook/script.js

```diff
@@ -0,0 +1,319 @@
+					},
+					{
+						name: 'series_master_id',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'start_date_time',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'subject',
+						type: 'STRING',
+						length: 255
+					}
+				]
+			},
+			{
+				name: 'calender_view_attendees',
+				columns: [
+					{
+						name: 'unique_id',
+						type: 'STRING',
+						length: 255,
+						primaryKey: true
+					},
+					{
+						name: 'parent_i_cal_u_id',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'root_i_cal_u_id',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'email_address_address',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'email_address_name',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'type',
+						type: 'STRING',
+						length: 255
+					}
+				]
+			},
+			{
+				name: 'my_events',
+				columns: [
+					{
+						name: 'i_cal_u_id',
+						type: 'STRING',
+						length: 255,
+						primaryKey: true
+					},
+					{
+						name: 'id',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'recurrence_pattern_day_of',
+						type: 'INTEGER'
+					},
+					{
+						name: 'recurrence_pattern_type',
+						type: 'STRING',
+						length: 255
+					},
+					{
+						name: 'recurrence_range_end_date',
+						type: 'DATE'
+					}
+				]
+			}
+		],
+		relationships: [
+			{
+				name: 'nested_table_1',
+				primaryTable: 'calender_view',
+				foreignTable: 'calender_view_attendees',
+				columnPairs: [
+					{
+						primaryKey: 'i_cal_u_id',
+						foreignKey: 'parent_i_cal_u_id'
+					}
+				]
+			},
+			{
+				name: 'fk_my_events_id',
+				primaryTable: 'calender_view',
+				foreignTable: 'my_events',
+				columnPairs: [
+					{
+						primaryKey: 'series_master_id',
+						foreignKey: 'id'
+					}
+				]
+			},
+			{
+				name: 'fk_calender_v_organizer_e',
+				primaryTable: 'users',
+				foreignTable: 'calender_view',
+				columnPairs: [
+					{
+						primaryKey: 'user_principal_name',
+						foreignKey: 'organizer_email_address_a'
+					}
+				]
+			}
+		]
+	}
+});
+
+const moment = library.load('moment-timezone');
+const uuid = library.load('uuid');
+const startDate = moment.utc().subtract(1, 'd').format();
+const endDate = moment.utc().add(30, 'd').format();
+const pageSize = 100;
+let userId = [];
+
+async function fullSync({ client, dataStore }) {
+	let URL = `/v1.0/users?$top=${pageSize}`;
+	let nextPage = '';
+	do {
+		const userRequest = await client.fetch(URL);
+		if (!userRequest.ok) {
+			throw new Error(`Users sync failed ${userRequest.status}:${userRequest.statusText}.`);
+		}
+		const userResponse = await userRequest.json();
+		for (const value of userResponse.value) {
+			userId.push(value.id);
+			dataStore.save('user', {
+				id: value.id,
+				mail: value.mail,
+				user_principal_name: value.userPrincipalName,
+				display_name: value.displayName
```

Comment:
```suggestion
				id: value?.id ?? null,
				mail: value?.mail ?? null,
				user_principal_name: value?.userPrincipalName ?? null,
				display_name: value?.displayName ?? null
```
We need to handle the undefined parameters in the response 
Please implement it for rest of the functions.

Reply:
Make sense!
Updating the code accordingly.

Reply:
It's not mandatory, it is just a good practice to handle the undefined parameters

Reply:
Gotcha! okay will update as suggested.